src/database/RecipeEntity.js

```diff
@@ -1,3 +1,84 @@
 const {Datastore} = require("@google-cloud/datastore");
 const datastore = new Datastore();
 
+async function getRecipeEntities(pageCursor) {
+  let query = datastore.createQuery("Recipe");
+
+  if (pageCursor) {
+    query = query.start(pageCursor);
+  }
+
+  const results = await datastore.runQuery(query);
+  const entities = results[0];
+  const info = results[1];
+
+  if (info.moreResults !== Datastore.NO_MORE_RESULTS) {
+    const results = await getRecipes(info.endCursor);
+    return [...entities, results[0]];
+  }
+  return entities;
+}
+
+async function getRecipes(data) {
+  if (!data.recipes) {
+    const entities = await getRecipeEntities();
+    const obj = entities.map(recipe => {
+      return {
+        id: recipe[datastore.KEY].id,
+        name: recipe.name,
+        ingredients: recipe.ingredients,
+        method: recipe.method,
+        createdBy: recipe.createdBy,
+        time: recipe.time
+      };
+    });
+    data.recipes = JSON.stringify(obj);
+  }
+  let recipes = JSON.parse(data.recipes);
```

Comment:
I don't understand this - you stringify the recipe objects, and then you parse it again? Why not just leave them in data format?